canonical/maas-ui
# feat(machines): paginate using the new API

## Done

- Add pagination to the machine list using the new API.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

To test this you either need more than one page of machines using [sampledata](https://github.com/canonical/maas-ui/blob/main/docs/HACKING.md#sample-data) or change [`const pageSize = DEFAULT_PAGE_SIZE;`](https://github.com/canonical/maas-ui/pull/4344/files#diff-2b86f4686bb74434cdb1fafb6f4ee8837c1cfa1afaf9d92934a85ee1a90072c8R56) in `MachineList.tsx` to something smaller than the number of machines you have.
- Go to the machine list.
- Scroll down and should see the pagination.
- Open the redux dev tools and go to `machine` -> `lists` and you should see a few requests.
- Clicking on the pagination buttons should navigate through the list and you should see one of the requests getting removed and replaced with a new one.

## Fixes

Fixes: canonical/app-tribe#1101.
Fixes: canonical/app-tribe#1121.
